File: tests/example_module.cpp
// Copyright (C) by Josh Blum. See LICENSE.txt for licensing information.

#include <gras/block.hpp>
#include <gras/factory.hpp>
#include <iostream>

struct MyBlock : gras::Block
{
    MyBlock(void):
        gras::Block("MyBlock")
    {
        this->register_call("get_num", &MyBlock::get_num);
    }

    ~MyBlock(void)
    {
        std::cout << "~MyBlock" << std::endl;
    }

    int get_num(void)
    {
        return 42;
    }

    //dummy work
    void work(const InputItems &, const OutputItems &){}
};

GRAS_REGISTER_FACTORY0("/tests/my_block0", MyBlock)

File: tests/example_module.py
# Copyright (C) by Josh Blum. See LICENSE.txt for licensing information.

import gras

class MyBlock(gras.Block):
    def __init__(self):
        gras.Block.__init__(self, "MyBlock")
        self.foo = 0
        self.register_call("get_num", self.get_num)

    def work(self, *args): pass

    def get_num(self):
        return 42

gras.register_factory("/tests/my_block1", MyBlock)
